Add Secure Login & Registration with Password Hashing

RishiDoshi3103 opened the pull request:
- Password Salting & Hashing: Implemented secure storage of user passwords using salts and SHA-256 hashing.

- Login Flow: Updated the login logic to verify hashed passwords, ensuring better protection against brute force attacks.

- Registration: Refined user registration to immediately hash and store new passwords securely.
 
- Validation & Error Handling: Added basic validation checks to handle invalid or duplicate usernames, and clear error messages if registration fails.
 
- Testing: Verified new features through unit tests (e.g., UserManagerTest) to ensure correctness and prevent regressions.

Commit:
Added base music player functionality. Added Player.java, Track.java, and related tests. Minor adjustments to TextView.java, LibraryModel.java. Little more testing/implementation to do with top 10 played songs queue, and 10 recent linked list variables.

Commit:
Changed library of songs to hashmap (value is play counts), added functions to help 10 freq/Recent playlists, added playSong/updateMostplayed to LibraryModel, updated textview to include plays/changes (Removed player/Playertest/track - unnecessary/adjusted

Commit:
Extra file clean up

Commit:
Additional unecessary file removal, added test to LibraryModelTest for song play/frequent/recent

RishiDoshi3103 committed:
Created User Securities

RishiDoshi3103 commented:
It's not perfect; I am still working on it. But it's something you can see.

Azkbecker commented:
Looks like good progress so far! I'm still working on the player_final branch, and have added: the extra sorted lists, and song/album removal.

 I'll likely have shuffling done by the end of tonight too, and look to have them added to my pull request by tomorrow. I'll probably be able to have the rest done (or almost) by tomorrow night, bugs notwithstanding, and we can meet in the afternoon Saturday / Sunday - depending how it all goes, and your availability of course - for a video / end report meeting!

Commit:
Added Song sorting, removing Song/Album, and Song/Playlist shuffle

RishiDoshi3103 commented:
Sure thing, we can meet tomorrow and you can decide when we can meet up.

On Thu, Mar 20, 2025 at 8:27 PM Kyle Becker ***@***.***>
wrote:

> Looks like good progress so far! I'm still working on the player_final
> branch, and have added: the extra sorted lists, and song/album removal.
>
> I'll likely have shuffling done by the end of tonight too, and look to
> have them added to my pull request by tomorrow. I'll probably be able to
> have the rest done (or almost) by tomorrow night, bugs notwithstanding, and
> we can meet in the afternoon Saturday / Sunday - depending how it all goes,
> and your availability of course - for a video / end report meeting!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/RishiDoshi3103/CSC_335-Project/pull/21#issuecomment-2742155221>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXXOFH6A4YDMTVVDTMF62AT2VOBKTAVCNFSM6AAAAABZO5UWN6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONBSGE2TKMRSGE>
> .
> You are receiving this because you were assigned.Message ID:
> ***@***.***>
> [image: Azkbecker]*Azkbecker* left a comment
> (RishiDoshi3103/CSC_335-Project#21)
> <https://github.com/RishiDoshi3103/CSC_335-Project/pull/21#issuecomment-2742155221>
>
> Looks like good progress so far! I'm still working on the player_final
> branch, and have added: the extra sorted lists, and song/album removal.
>
> I'll likely have shuffling done by the end of tonight too, and look to
> have them added to my pull request by tomorrow. I'll probably be able to
> have the rest done (or almost) by tomorrow night, bugs notwithstanding, and
> we can meet in the afternoon Saturday / Sunday - depending how it all goes,
> and your availability of course - for a video / end report meeting!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/RishiDoshi3103/CSC_335-Project/pull/21#issuecomment-2742155221>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXXOFH6A4YDMTVVDTMF62AT2VOBKTAVCNFSM6AAAAABZO5UWN6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONBSGE2TKMRSGE>
> .
> You are receiving this because you were assigned.Message ID:
> ***@***.***>
>

RishiDoshi3103 committed:
Made some Twiks for Test

Make some changes so runs perfectly and now every package is above 90% except View Package

RishiDoshi3103 committed:
Made Multiple User And Store the data in the User

RishiDoshi3103 commented:
![Screenshot 2025-03-21 154439](https://github.com/user-attachments/assets/3d829569-c3d4-4248-a376-e4934215db12)

It Succefully Loaded  the user files.

RishiDoshi3103 committed:
Debug the UserSecurity

Commit:
Added song removal to Album.java. Added favorite songs/top rated playlists, song removal, album removal, add album with one song or all, add song to existing album, and song search by genre to LibraryModel.java. Removed tentative genre comments from Song.java. Adjusted TextView.java to accommodate above, including functionalitites.

RishiDoshi3103 committed:
Merge branch 'User_Security_Functions_Final' into player_final

Azkbecker committed:
Merge pull request #22 from RishiDoshi3103/player_final

Approved!

RishiDoshi3103 committed:
Fix Bugs
